### PR TITLE
fixing IDENTITY-4105

### PR DESCRIPTION
**Pull request by GayanM**

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ private void addPasswordContainerToCache(RandomPassword[] randomPasswords, Strin
         RandomPasswordContainerCache.getInstance().addToCache(cacheKey, cacheEntry);
     }
 
-    private RandomPassword[] getAndRemoveRandomPasswordContainerFromCache(String uuid) {
+    private RandomPassword[] getAndRemoveRandomPasswordContainerFromCache(String uuid, boolean withCacheClear) {
```
Comment (darshanasbg) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java]: method name should be changed as with the new parameter its not always clearing the entry

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java [hunk 2]
```diff
@@ -156,8 +156,9 @@ private void addPasswordContainerToCache(RandomPassword[] randomPasswords, Strin
         if (log.isDebugEnabled()){
             log.debug("Removing Cache Entry with Key: " + uuid);
```
Comment (darshanasbg) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java]: This log should be moved inside the next if block.

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java [hunk 3]
```diff
@@ -101,7 +101,7 @@ public static RandomPasswordProcessor getInstance(){
                 log.debug("Cache Key not found for Random Password Container");
             }
```
Comment (darshanasbg) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java]: Why need to cast to Property[] above?
````java
properties = (Property[]) removeUniqueIdProperty(properties);
`````

**Comment by GayanM** [comment]
please on hold the pull request

**Pull request merged**